android/src/main/java/com/dotlottiereactnative/DotlottieReactNativeViewManager.kt

```diff
@@ -1,50 +1,137 @@
+            "onLoop"
+    )
+  }
+
+  override fun receiveCommand(
+          view: DotlottieReactNativeView,
+          commandId: Int,
+          args: ReadableArray?
+  ) {
+
+    if (commandId == COMMAND_PLAY_ID) view.dotLottieController.play()
+    if (commandId == COMMAND_PAUSE_ID) view.dotLottieController.pause()
+    if (commandId == COMMAND_STOP_ID) view.dotLottieController.stop()
+    if (commandId == COMMAND_SET_SPEED_ID) {
+      val speed = args?.getDouble(0)?.toFloat() ?: 1f
+      view.dotLottieController.setSpeed(speed)
+    }
+    if (commandId == COMMAND_FREEZE_ID) {
+      view.dotLottieController.freeze()
+    }
+    if (commandId == COMMAND_UNFREEZE_ID) {
+      view.dotLottieController.unFreeze()
+    }
+    if (commandId == COMMAND_SET_LOOP_ID) {
+      val loop = args?.getBoolean(0) ?: false
+      view.dotLottieController.setLoop(loop)
+    }
+
+    if (commandId == COMMAND_SET_FRAME_ID) {
+      val frame = args?.getDouble(0)?.toFloat() ?: 0f
+      view.dotLottieController.setFrame(frame)
+    }
+    if (commandId == COMMAND_SET_PLAY_MODE_ID) {
+      val mode = args?.getInt(0)
+      val modeValue = Mode.values()[mode ?: 0]
+      view.dotLottieController.setPlayMode(modeValue)
     }
   }
 
   @ReactProp(name = "source")
-  fun setSource(view: ComposeView, url: String?) {
-    // Update the URL and re-compose the view
-    animationUrl = url
-    view.setContent { DotLottieContent() }
+  fun setSource(view: DotlottieReactNativeView, url: String?) {
+    view.setSource(url)
   }
 
   @ReactProp(name = "loop")
-  fun setLoop(view: ComposeView, value: Boolean) {
-    loop = value
-    view.setContent { DotLottieContent() }
+  fun setLoop(view: DotlottieReactNativeView, value: Boolean) {
+    view.setLoop(value)
   }
 
   @ReactProp(name = "autoplay")
-  fun setAutoPlay(view: ComposeView, value: Boolean) {
-    autoplay = value
-    view.setContent { DotLottieContent() }
+  fun setAutoPlay(view: DotlottieReactNativeView, value: Boolean) {
+    view.setAutoPlay(value)
+  }
+
+  @ReactProp(name = "speed")
+  fun setSpeed(view: DotlottieReactNativeView, value: Double) {
+    view.setSpeed(value)
+  }
+
+  companion object {
+    const val TAG = "DotlottieReactNativeViewManager"
+
+    private const val COMMAND_PLAY = "play"
+    private const val COMMAND_PLAY_ID = 1
+
+    private const val COMMAND_PAUSE = "pause"
+    private const val COMMAND_PAUSE_ID = 2
+
+    private const val COMMAND_STOP = "stop"
+    private const val COMMAND_STOP_ID = 3
+
+    private const val COMMAND_SET_SPEED = "setSpeed"
+    private const val COMMAND_SET_SPEED_ID = 4
+
+    private const val COMMAND_FREEZE = "freeze"
+    private const val COMMAND_FREEZE_ID = 5
+
+    private const val COMMAND_UNFREEZE = "unFreeze"
+    private const val COMMAND_UNFREEZE_ID = 6
+
+    private const val COMMAND_SET_LOOP = "setLoop"
+    private const val COMMAND_SET_LOOP_ID = 7
+
+    private const val COMMAND_SET_PROGRESS = "setProgress"
```

Comment:
what is this command is used for ?

Reply:
Will remove this. was trying to use it for segment